### PR TITLE
chore: Support local CMC in e2e tests

### PR DESCRIPTION
**Pull request by adamspofford-dfinity**
This patches the cycles minting canister before installing it as part of the ledger package, so the XDR:ICP rate can be set. The method used is copied from nns-dapp. Unfortunately *creating* a canister requires a subnet, which the local replica doesn't have, so for now we can only test cycle *top-up*.

Resolves SDK-514.

### CODE REVIEW

**Comment by sesi200** [comment]
LGTM, but do we have a strategy to keep this at least somewhat up-to-date?

**Comment by adamspofford-dfinity** [comment]
We would simply update the IC commit here when we do it elsewhere as well.

File: scripts/update-replica.sh [hunk 1]
```diff
@@ -26,4 +26,9 @@ niv update sandbox-launcher-x86_64-linux -a rev=$SHA
 echo "Writing asset sources"
 ./scripts/write-dfx-asset-sources.sh
 
+read -r -p "Update the ledger for e2e tests? (y/n)" yn
+if [[ ! $yn =~ [nN] ]]; then
+    ./scripts/update-nns.bash "$SHA"
```
Comment (sesi200) [inline scripts/update-replica.sh:31]: Does this also update nns-canister.patch? I don't see how the patch file is staying updated.

Comment (sesi200) [reply scripts/update-replica.sh:31]: I think this is it: https://github.com/dfinity/nns-dapp/blob/main/e2e-tests/scripts/nns-canister.patch

**Review by sesi200 — APPROVED** [review]

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**